### PR TITLE
BUG: switch _CMP_NEQ_OQ to _CMP_NEQ_UQ for npyv_cmpneq_f[32,64]

### LINKED ISSUE #21685: BUG: Incorrect handling of not-equal comparison to nan
### Describe the issue:

For a sufficiently large array `x`, the comparison `x != 0` returns False for elements of `x` that are `nan`. The result should be True.  For example,
```
In [1]: import numpy as np

In [2]: x = np.zeros((4, 8))

In [3]: x[0,0] = np.nan

In [4]: x != 0  # The first element of the result should be True
Out[4]: 
array([[False, False, False, False, False, False, False, False],
       [False, False, False, False, False, False, False, False],
       [False, False, False, False, False, False, False, False],
       [False, False, False, False, False, False, False, False]])

In [5]: x[:2] != 0  # The result with a smaller array works as expected.
Out[5]: 
array([[ True, False, False, False, False, False, False, False],
       [False, False, False, False, False, False, False, False]])

```
The bug appears to have been introduced in gh-21483.

### NumPy/Python version information:

```
In [6]: import sys, numpy; print(numpy.__version__, sys.version)
1.24.0.dev0+136.g15b92f7ab 3.10.1 (main, Jan 14 2022, 02:27:20) [GCC 11.2.0]
```

### PR DESCRIPTION
**Pull request by rafaelcfsousa**
This PR fixes #21685.

The comparison function `np.equal_not` returns `True` when one of the operands is `NaN`. The other comparison functions, on the other hand, return `False`.

cc: @seberg

### CODE REVIEW

**Comment by seberg** [comment]
@rafaelcfsousa do we have tests that notice the problem (but maybe not on our CI?).  If not, it would be best to add tests as part of this PR.

**Comment by rafaelcfsousa** [comment]
No, we do not have a test able to notice this issue. I will add a test in `numpy/core/tests/test_simd.py`.

**Review by seberg — APPROVED** [review]
LGTM, I am not too familiar with this simd test file though, so going to hold off merging in case anyone has a suggestion.

I wouldn't mind having integration tests in `test_umath` or so, that just put in longer (odd sized) vectors, so that we definitely also cover any potential peeling behavior.  But this seems sufficient here.

File: numpy/core/tests/test_simd.py [hunk 1]
```diff
@@ -501,6 +501,34 @@ def test_special_cases(self):
...
+    @pytest.mark.parametrize('py_comp,np_comp', [
+        (operator.lt, "cmplt"),
+        (operator.le, "cmple"),
+        (operator.gt, "cmpgt"),
+        (operator.ge, "cmpge"),
+        (operator.eq, "cmpeq"),
+        (operator.ne, "cmpneq")
+    ])
+    def test_comparison_with_nan(self, py_comp, np_comp):
+        pinf, ninf, nan = self._pinfinity(), self._ninfinity(), self._nan()
```
Comment (seberg) [inline numpy/core/tests/test_simd.py:515]: Hmmm, I have a question after all.  This seems like it tests float64 only?  Whether or not we add more tests, I think this should cover all tests.

Or... add generic float tests to `test_umath.py` if it doesn't fit here?

Comment (rafaelcfsousa) [reply numpy/core/tests/test_simd.py:515]: It also covers float32

```numpy/core/tests/test_simd.py::Test_SIMD_FP_128_baseline_f32::test_comparison_with_nan[lt-cmplt] PASSED                                                                                                                                [ 83%] 
numpy/core/tests/test_simd.py::Test_SIMD_FP_128_baseline_f32::test_comparison_with_nan[le-cmple] PASSED                                                                                                                                [ 83%] 
numpy/core/tests/test_simd.py::Test_SIMD_FP_128_baseline_f32::test_comparison_with_nan[gt-cmpgt] PASSED                                                                                                                                [ 83%] 
numpy/core/tests/test_simd.py::Test_SIMD_FP_128_baseline_f32::test_comparison_with_nan[ge-cmpge] PASSED                                                                                                                                [ 83%] 
numpy/core/tests/test_simd.py::Test_SIMD_FP_128_baseline_f32::test_comparison_with_nan[eq-cmpeq] PASSED                                                                                                                                [ 83%] 
numpy/core/tests/test_simd.py::Test_SIMD_FP_128_baseline_f32::test_comparison_with_nan[ne-cmpneq] PASSED    ```

Comment (seberg) [reply numpy/core/tests/test_simd.py:515]: Ah, thanks... inheritance magic I guess, sorry for the noise.

**Comment by seberg** [comment]
Thanks for the super-fast followup @rafaelcfsousa!  If anyone has suggestions about tests, please don't  hesitate to comment.

**Pull request merged**